pytest_splunk_addon/standard_lib/fields_tests/test_templates.py

```diff
@@ -405,21 +408,34 @@ def test_datamodels(
         extracted_tags = fields_from_splunk.get("tag", "")
         extracted_tags = extracted_tags.strip("][").split(", ")
         extracted_tags = [tag.replace("'", "") for tag in extracted_tags]
-
-        extracted_datamodels = []
-        for datamodel, tags in dict_datamodel_tag.items():
-            if all(tag in extracted_tags for tag in tags):
-                extracted_datamodels.append(datamodel)
-
         self.logger.debug(f"Tags extracted from Splunk {extracted_tags}")
 
+        matched_datamodels = {
+            dm: tags
+            for dm, tags in dict_datamodel_tag.items()
+            if all(tag in extracted_tags for tag in tags)
+        }
+        assigned_datamodels = {
+            dm: tags
+            for dm, tags in matched_datamodels.items()
+            if not any(
+                set(tags).issubset(set(matched_tags)) and dm != matched_datamodel
+                for matched_datamodel, matched_tags in matched_datamodels.items()
+            )
+        }
+
         record_property("search", search)
 
-        missing_datamodels = [dm for dm in datamodels if dm not in extracted_datamodels]
-        wrong_datamodels = [dm for dm in extracted_datamodels if dm not in datamodels]
+        missing_datamodels = {
+            dm: tags for dm, tags in datamodels.items() if dm not in assigned_datamodels
+        }
+        wrong_datamodels = {
+            dm: tags for dm, tags in assigned_datamodels.items() if dm not in datamodels
+        }
 
         self.logger.debug(
-            f"List of missing tags {missing_datamodels}. List of wrongly assigned tags {wrong_datamodels}"
+            f"List of missing datamodels {missing_datamodels.keys()} and tags {[chain.from_iterable(missing_datamodels.values())]}"
```

Comment:
Is it always true? For example if only one of two tags from datamodel is missing? Then we will log one tag which is not missing too?

Reply:
maybe it should be only logged if there is sth missing or wrong? Then it can be logged as info or warn